### PR TITLE
Add Mezzanine

### PR DESCRIPTION
**Pull request by cadecairos**

### CODE REVIEW

File: app/networkapi/opportunities/models.py [hunk 1]
```diff
@@ -13,7 +12,7 @@ def get_opportunities_image_upload_path(instance, filename):
     )
 
 
-class Opportunity(SortableMixin):
+class Opportunity(models.Model):
```
Comment (Pomax) [inline app/networkapi/opportunities/models.py]: did this not work with mezzanine?

File: app/networkapi/settings.py [hunk 2]
```diff
@@ -45,7 +46,9 @@
 # SECURITY WARNING: don't run with debug turned on in production!
 DEBUG = env('DEBUG')
 
-ALLOWED_HOSTS = env('ALLOWED_HOSTS')
+ALLOWED_HOSTS = os.getenv("ALLOWED_HOSTS").split(",")
```
Comment (Pomax) [inline app/networkapi/settings.py]: if you're using getenv, you can make this safer as `ALLOWED_HOSTS = os.getenv("ALLOWED_HOSTS", '').split(",")` so that the fallback value is guaranteed safe

File: requirements.txt [hunk 3]
```diff
@@ -1,8 +1,11 @@
 # Core
-Django==1.10.3
+Django==1.10.6
```
Comment (Pomax) [inline requirements.txt]: how do you keep this so nicely formatted... `pip freeze` just outputs an alphabeticallysorted list with all comments and whitespace removed O_o

File: app/networkapi/opportunities/models.py (hunk 1 above)
Comment (cadecairos) [inline app/networkapi/opportunities/models.py]: Doing it in a separate PR

File: requirements.txt (hunk 3 above)
Comment (cadecairos) [inline requirements.txt]: I wasn't aware of pip freeze until now :)

Comment (Pomax) [inline requirements.txt]: I seeeee

File: app/networkapi/settings.py (hunk 2 above)
Comment (gideonthomas) [inline app/networkapi/settings.py]: You really shouldn't need to make this change. `env('ALLOWED_HOSTS')` is meant to do that. Otherwise it's just not configured properly

File: app/networkapi/settings.py [hunk 4]
```diff
@@ -54,52 +57,88 @@
...
+)
+
 ROOT_URLCONF = 'networkapi.urls'
 
+#########
+# PATHS #
+#########
+
+# Full filesystem path to the project.
+PROJECT_APP_PATH = os.path.dirname(os.path.abspath(__file__))
```
Comment (gideonthomas) [inline app/networkapi/settings.py]: use `root(path)` and `app(path)` accordingly instead

File: app/networkapi/settings.py [hunk 5]
```diff
@@ -54,52 +57,88 @@
...
+PROJECT_APP = os.path.basename(PROJECT_APP_PATH)
+PROJECT_ROOT = BASE_DIR = os.path.dirname(PROJECT_APP_PATH)
+
 TEMPLATES = [
     {
         'BACKEND': 'django.template.backends.django.DjangoTemplates',
         'DIRS': [
-            app('templates'),
-        ],
+            os.path.join(PROJECT_ROOT, "templates")
```
Comment (gideonthomas) [inline app/networkapi/settings.py]: why didn't this work?

File: env.default [hunk 6]
```diff
@@ -9,3 +9,4 @@ CONTENT_TYPE_NO_SNIFF=True
 SET_HSTS=False
 SSL_REDIRECT=False
 X_FRAME_OPTIONS=DENY
+ALLOWED_HOSTS=localhost
```
Comment (gideonthomas) [inline env.default]: blank line

File: app/networkapi/settings.py (hunk 2 above)
Comment (cadecairos) [inline app/networkapi/settings.py]: It's not configured properly. The variable is interpreted as a tring instead of a list, so it just doesn't work.

File: app/networkapi/settings.py (hunk 4 above)
Comment (cadecairos) [inline app/networkapi/settings.py]: Can you elaborate on this?

File: app/networkapi/settings.py (hunk 2 above)
Comment (gideonthomas) [inline app/networkapi/settings.py]: I'm literally not able to reproduce this. I always get a list

File: app/networkapi/settings.py (hunk 4 above)
Comment (cadecairos) [inline app/networkapi/settings.py]: do you mean `PROJECT_APP_PATH = root(__file__)`?

File: app/networkapi/settings.py (hunk 2 above)
Comment (gideonthomas) [inline app/networkapi/settings.py]: I would like to stick to one `env` pattern instead of two and `env` is a really good library for that. Maybe let's try to debug over vidyo?

File: app/networkapi/settings.py (hunk 5 above)
Comment (cadecairos) [inline app/networkapi/settings.py]: not sure. I can revert to double check.

File: app/networkapi/settings.py (hunk 2 above)
Comment (Pomax) [inline app/networkapi/settings.py]: using `env.ALLOWED_HOSTS` has no fallback safety and will **crash** execution if there is no environment variable by you're telling python to access a nonexistent property. To get around this, you can use a dictionary-safe access like `oi.getenv` instead, which will not crash python, but will give you `None` back if there is no environment variable. Given that we immediately call `split` on the result, and `None.split()` again **does** crash python, providing that fallback value explicitly as a string is super nice.

Comment (cadecairos) [inline app/networkapi/settings.py]: I'm pretty sure `env` is the reason I can't run this app locally on windows (I can only run it in docker)
I can run the coss app no problem on windows though, so I feel like the `env` pattern might cater to UNIX like systems only.

Comment (gideonthomas) [inline app/networkapi/settings.py]: we provide a default above as an empty list

File: app/networkapi/settings.py (hunk 4 above)
Comment (gideonthomas) [inline app/networkapi/settings.py]: what path are you trying to get? It looks like just `app` will suffice.

File: app/networkapi/settings.py [hunk 7]
```diff
@@ -54,52 +57,88 @@
...
+PROJECT_APP = os.path.basename(PROJECT_APP_PATH)
+PROJECT_ROOT = BASE_DIR = os.path.dirname(PROJECT_APP_PATH)
+
 TEMPLATES = [
     {
         'BACKEND': 'django.template.backends.django.DjangoTemplates',
         'DIRS': [
-            app('templates'),
-        ],
+            app('templates')
```
Comment (gideonthomas) [inline app/networkapi/settings.py]: why this change?

File: app/networkapi/settings.py (hunk 2 above)
Comment (Pomax) [inline app/networkapi/settings.py]: I can use `environ` on my windows machine, so that sounds like it might be a different problem

File: app/networkapi/settings.py (hunk 7 above)
Comment (cadecairos) [inline app/networkapi/settings.py]: I was reverting an earlier change, forgot a trailing comma. I'll add it back in

File: app/networkapi/urls.py [hunk 8]
```diff
@@ -15,16 +15,25 @@
...
 urlpatterns = [
-    url(r'^admin/', admin.site.urls),
-    url(r'^people/', include('networkapi.people.urls')),
-    url(r'^opportunities/', include('networkapi.opportunities.urls')),
-    url(r'^news/', include('networkapi.news.urls')),
+    url("^admin/", include(admin.site.urls)),
+    url(r'^api/people/', include('networkapi.people.urls')),
+    url(r'^api/opportunities/', include('networkapi.opportunities.urls')),
+    url(r'^api/news/', include('networkapi.news.urls')),
+    url("^$", mezzanine.pages.views.page, {"slug": "/"}, name="home"),
```
Comment (gideonthomas) [inline app/networkapi/urls.py]: nit: single quotes everywhere

**Comment by Pomax** [comment]
Python has a well established linter, conformant PEP-8,  it sounds like it's time to add a linting requirement to Travis here so we don't have manually fix up commits =)

**Comment by Pomax** [comment]
awesome
